source/manual/migrate-to-dart-sass-from-libsass.html.md

```diff
@@ -0,0 +1,205 @@
+
+```diff
+- config.sass.style = :compressed
+- config.sass.line_comments = false
+  ...
+```
+
+## Builds
+
+`dartsass:build` is linked to `assets:precompile`. When `assets:precompile` is run, it triggers the compilation of Sass files, which are generated in the `/app/assets/builds` directory. They are then processed by Sprockets.
+
+## Configuring build options
+
+See [configuring build options](https://github.com/rails/dartsass-rails#configuring-build-options) and [Dart Sass Command-Line Interface](https://sass-lang.com/documentation/cli/dart-sass/)
+
+## Watch mode
+
+When you're developing your application, you can run Sass in watch mode, so stylesheets are continuously updated when Sass files are changed.
+
+1. Add `bin/dev`
+
+   ```ruby
+   #!/usr/bin/env sh
+   if ! gem list foreman -i --silent; then
+     echo "Installing foreman..."
+     gem install foreman
+   fi
+   exec foreman start -f Procfile.dev "$@"
+   ```
+
+2. Add (or update) `procfile.dev`
+
+   ```ruby
+   web: bin/rails server -p 3070
+   css: bin/rails dartsass:watch
+   ```
+
+3. Update `startup.sh`
+
+   ```diff
+   if [[ $1 == "--live" ]] ; then
+     GOVUK_APP_DOMAIN=www.gov.uk \
+     GOVUK_WEBSITE_ROOT=https://www.gov.uk \
+     GOVUK_PROXY_STATIC_ENABLED=true \
+     PLEK_SERVICE_CONTENT_STORE_URI=${PLEK_SERVICE_CONTENT_STORE_URI-https://www.gov.uk/api} \
+     PLEK_SERVICE_STATIC_URI=${PLEK_SERVICE_STATIC_URI-https://assets.publishing.service.gov.uk} \
+     PLEK_SERVICE_SEARCH_API_URI=${PLEK_SERVICE_SEARCH_API_URI-https://www.gov.uk/api} \
+   - bundle exec rails s -p 3070
+   + ./bin/dev
+   else
+     ...
+   ```
+
+4. In Docker, update `docker-compose.yml`
+
+   ```diff
+     expose:
+   -  - "3000"
+   - command: bin/rails s --restart
+   +  - "3070"
+   + command: ./bin/dev
+     ...
+   ```
+
+### Digests
+
+To see the latest stylesheet changes when developing your application, if running Sass in watch mode, you'll have to turn off digests. Update `config/environments/development.rb` to add
+
+```ruby
+  config.assets.digest = false
+```
+
+See [3.2 Turning Digests Off - The Asset Pipeline — Ruby on Rails Guides](https://guides.rubyonrails.org/asset_pipeline.html#turning-digests-off).
+
+## Troubleshooting
+
+See [common issues](https://github.com/rails/dartsass-rails#troubleshooting).
+
+### Other issues
+
+#### Globs
+
+Any form of pattern matching added for importing Sass files will not work. Sass files needs to be imported individually.
+
+```diff
+- @import "modules/*";
++ @import "modules/module-1";
++ @import "modules/module-2";
++ @import "modules/module-3";
++ @import "modules/module-4";
+  ...
+```
+
+#### Asset paths
+
+Replace occurrences of the `image-url` and `asset-url` helpers with the `url()` CSS function.
+
+```diff
+.my-image {
+- background-image: image-url("path/to/my/image.svg");
++ background-image: url("path/to/my/image.svg");
+  background-color: transparent;
+  ...
+}
+```
+
```

Comment:
This could be a good place to include further information on how to set custom image / font paths using the customer image / font handlers:

- [Custom image and font handler](https://github.com/alphagov/govuk_publishing_components/pull/3604/files#diff-90b6290ad9dc7dc57ff037f612539075a553420a9dd69092ff75c62d677cae6f)
- [Example of usage in static](https://github.com/alphagov/static/compare/main...test-dart-sass#diff-0869e3045a3b5ca0e44ed193977d1568d686ca2373f4b93a8a868e4099fc932dR1-R2)

Reply:
Could you remind me why we need to do this outside of the gem? Just for the apps it seems you only need to replace the Rails helpers and don't need any additional config. I can't quite remember why we needed to include custom paths.